### ISSUE TITLE
default('') === undefined but should be an empty string

### ISSUE THREAD

**Issue by joh-klein** [evanshortiss/env-var#101]
**Describe the bug 🐛**
If `default` is an empty string `undefined` is returned.

**To Reproduce 📝**
```js
const env = require('env-var')

// Use POOL_SIZE if set, else use an empty value
const POOL_SIZE = env.get('POOL_SIZE').default('').asString()
```
**Expected behaviour 🤷‍♂️🤷**
`POOL_SIZE === ''` should be true, not false

**Environment (please complete the following information) 💻:**
 - OS: macOS 10.15.2
 - Runtime node.js
 - Version 12.16.1

**Comment by evanshortiss** [comment]
This is a strange one. 

Do you think there's a valid reason for setting default to an empty string instead of undefined? 

Both are false-like values so doing things like below would work regardless.

```
if (!env.get('POOL_SIZE').default('').asString()) {
  // Default empty string means we'll always get here
  // unless it was set to a non-empty string in the environment
}
```

Would like to know your thoughts 😀

**Comment by joh-klein** [comment]
Thanks for looking into it. I will gladly share my thoughts:
We are mainly using env-var in the configuration of our node servers. And part of that configuration are paths (url paths and file paths). So for example

```js
// relative vs absolute file paths
const BASE_PATH = env.get('BASE_PATH').default('').asString();

path.join(BASE_PATH, 'path', 'to', 'srcDir');
// Current: either '/var/app/path/to/srcDir' or 'TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received undefined'
// Expected: either  '/var/app/path/to/srcDir' or  'path/to/srcDir'
```
```js
// relative vs absolute urls 
const BASE_PATH = env.get('BASE_PATH').default('').asString();

const imgSrc = `${BASE_PATH}/path/to/logo.svg`;
// Current: either 'https://domain.tld/path/to/logo.svg' or 'undefined/path/to/logo.svg'
// Expected: either 'https://domain.tld/path/to/logo.svg' or '/path/to/logo.svg'
```

Two other things: 
- I would always expect, that `.asString()` would return a type string.
- I would ALWAYS expect, that, whatever I put into `.default()`, will be returned back to me (with in reason) (e.g. `.default(null) !== .default(undefined) !== .default('') !== .default(0) !== .default(false) !== .default('0') !== .default([]) …`)


BTW: My workaround right now is
```js
const POOL_SIZE = env.get('POOL_SIZE').default('').asString() || '';
```
but that defeats the beauty of the whole `.default()` setting. At least for me.

**Comment by evanshortiss** [comment]
> I would always expect, that .asString() would return a type string.

All env-var functions return either `Type` or `undefined`, where `Type` is String, Array, Object, Number or whatever is defined by the function being called, e.g `asString()`. I can't see a good reason to return an empty string for `asString()` if the variable is not set and no default is defined. Returning the empty string would then raise the question of  "do we return an empty object for `asJsonObject()` if a the target is not set - the answer to the latter is definitely "no" for me, so I feel that the former should also be "no".

**Comment by evanshortiss** [comment]
Fixed by #106
